src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -193,13 +192,26 @@ private void handleGetRecording(HttpExchange exchange, long id) {
                         });
     }
 
-    private void handleStart(HttpExchange exchange) throws IOException {
+    private void handleStartRecordingOrSnapshot(HttpExchange exchange) throws IOException {
         try (InputStream body = exchange.getRequestBody()) {
             StartRecordingRequest req = mapper.readValue(body, StartRecordingRequest.class);
             if (!req.isValid()) {
                 exchange.sendResponseHeaders(HttpStatus.SC_BAD_REQUEST, BODY_LENGTH_NONE);
                 return;
             }
+            if (req.requestSnapshot()) {
+                try {
+                    SerializableRecordingDescriptor snapshot = startSnapshot(req, exchange);
+                    exchange.sendResponseHeaders(HttpStatus.SC_CREATED, BODY_LENGTH_UNKNOWN);
+                    try (OutputStream response = exchange.getResponseBody()) {
+                        mapper.writeValue(response, snapshot);
+                    }
+                    return;
+                } catch (IOException e) {
+                    log.error("Failed to start snapshot", e);
+                    exchange.sendResponseHeaders(HttpStatus.SC_SERVICE_UNAVAILABLE, BODY_LENGTH_NONE);
+                }
```

Comment:
the `return` statement should either be at the end here, or the code lower down that calls into `startRecording` should be in an `else` so that this doesn't accidentally handle an IOException, send headers, and then continue to try to start a recording with an invalid request body.